cpp/src/io/parquet/writer_impl.cu

```diff
@@ -897,23 +942,22 @@ void writer::impl::init_page_fragments(cudf::detail::hostdevice_2dvector<gpu::Pa
   frag.device_to_host(stream, true);
 }
 
-void writer::impl::gather_fragment_statistics(
-  device_2dspan<statistics_chunk> frag_stats_chunk,
-  device_2dspan<gpu::PageFragment const> frag,
-  device_span<gpu::parquet_column_device_view const> col_desc,
-  uint32_t num_fragments)
+void writer::impl::init_page_fragments_1d(hostdevice_vector<gpu::PageFragment>& frag,
```

Comment:
Or should I move the `host_to_device()` call here and keep the signature?

Reply:
I'm leaning towards my version, it makes `init_page_fragments_1d` more flexible/reusable. But, since that does not really matter in this case, I don't have a strong opinion.